src/Microsoft.Extensions.Configuration.Abstractions/ConfigurationPath.cs

```diff
@@ -0,0 +1,45 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+
+namespace Microsoft.Extensions.Configuration
+{
+    /// <summary>
+    /// Utility methods and constants for manipulating Configuration paths
+    /// </summary>
+    public static class ConfigurationPath
+    {
+        public static readonly string KeyDelimiter = ":";
+
+        public static string Combine(params string[] pathSegements)
+        {
+            if (pathSegements == null)
+            {
+                throw new ArgumentNullException(nameof(pathSegements));
+            }
+            return string.Join(KeyDelimiter, pathSegements);
+        }
+
+        public static string Combine(IEnumerable<string> pathSegements)
+        {
+            if (pathSegements == null)
+            {
+                throw new ArgumentNullException(nameof(pathSegements));
+            }
+            return string.Join(KeyDelimiter, pathSegements);
+        }
+
+        public static string GetLastSegment(string path)
```

Comment:
Regardless of what we do with `Combine()` I think `GetLastSegment()` is less generally useful (e.g. we use it only in one place). 

I would be rather conservative about what we add here in the short term. We can always add more stuff here based on feedback.